### ISSUE TITLE
Image element blocks resizing

### ISSUE THREAD

**Issue by jmEvoqua** [slint-ui/slint#6865]
### Bug Description

During refactoring I noticed that an image element blocks the resizing(stretching) in a layout. If I replace the image with an rectangle of the same size the resizing works as expected.

![grafik](https://github.com/user-attachments/assets/272cf104-6152-4ce7-b66f-08f09d477468)

I expected a fixed sized image element to behave the same as a fixed sized rectangle:

![grafik](https://github.com/user-attachments/assets/9422b5f4-ffcb-40bd-967e-5016fbea6e18)

What makes it even worse is that when the image is conditional, it works as expected. This makes it very hard to reason about the layout system.

`if true: Image {`

![grafik](https://github.com/user-attachments/assets/361933ae-ea2e-4d1c-8576-8514c6763480)

### Reproducible Code (if applicable)

```slint
component Main inherits Window {

    HorizontalLayout {

        Rectangle {
            background: green;
        }

        Rectangle {
            background: red;

            Image {
                source: @image-url("https://slint.dev/logo/slint-logo-full-light.svg");
                colorize: white;
                width: 50px;
                height: 50px;
                
            }
        }
    }
}
```

### Environment Details

- Slint Version: 1.8.0
- Platform/OS: Linux
- Programming Language: Rust
- Backend/Renderer: 

### Product Impact

This sort of bugs makes it hard to understand the (not so intuitive) layout system and requires additional time to find workarounds. The initial fast development speed is slowed down drastically when something odd in the layout happens.

**Comment by tronical** [comment]
I dug a little further and found out what the difference is between using an image or a plain rectangle: An `Image` element always has its preferred size set to the natural size of the source. So if you have say an `Image { source: @image-url("foo.png"); }` where `foo.png` has a width of say `1000px`, then the `Image` will have a preferred width of `1000px` - in an attempt to express that the desired view is to show it unscaled.

So this is "reproducible" without an image:

```slint
component Main inherits Window {

    HorizontalLayout {

        Rectangle {
            background: green;
        }

        Rectangle {
            background: red;

            Rectangle {
                background: blue;
                width: 50px;
                height: 50px;
                preferred-width: 165px;
                preferred-height: 66px;
            }
        }
    }
}
```

(This is not a final statement of resolution, just posting my results of investigating this :)

**Comment by jmEvoqua** [comment]
Thanks for the help. This explains why the image has a starting width. But there is still one difference. 


The red rectangle with the image does not resize at all (not using more of the available space).

The red rectangle in your example with the other rectangle does resize. It will be larger than the green Rectangle, but that is expected.

**Comment by tronical** [comment]
That's right. I think that's because the plain `Rectangle` has a `horizontal-stretch: 1;` while the `Image` has `horizontal-stretch: 0;`. In the docs ("Stretch Algorithm") this is explained like this:

> The elements that have a default content size usually defaults to 0 while elements that default to the size of their parents defaults to 1.

**Comment by jmEvoqua** [comment]
Thanks for the explanation :)

I understand it now.

**Comment by tronical** [comment]
I think our initial reaction yesterday was to mark this as a bug. With this analysis, I think we need to improve on the documentation.

The central question was why the `Image` behaves differently than the `Rectangle`.

While the underlying information may be spread out in the docs, what's missing is a more element centric view on the behaviour of the layouts.

**Comment by tronical** [comment]
On second thought: Perhaps it's a bug that the Image's preferred size is used and propagated when the Image can never obtain that size because it's already fixed.
